libs/isograph-react/src/core/makeNetworkRequest.ts

```diff
@@ -166,13 +167,30 @@ export function makeNetworkRequest<
 
       const root = { __link: ROOT_ID, __typename: artifact.concreteType };
       if (status.kind === 'UndisposedIncomplete') {
-        normalizeData(
-          environment,
-          normalizationAst.selections,
-          networkResponse.data ?? {},
-          variables,
-          root,
+        const encounteredIds: EncounteredIds = new Map();
+        environment.store = addNetworkResponseStoreLayer(
```

Comment:
Since we're returning `storeLayer` (line 173), it would be simpler to do:

```
// returns either environment.store or a new node that points to it
const newChildStoreLayer = ensureChildStoreLayerAcceptsData(environment.store);
normalizeData(newChildStoreLayer);
environment.store = newChildStoreLayer;
```

Sorry, I know I suggested what you have here!